numpy/_core/src/multiarray/npy_static_data.c

```diff
@@ -0,0 +1,269 @@
+/* numpy static data structs and initialization */
+#define NPY_NO_DEPRECATED_API NPY_API_VERSION
+#define _UMATHMODULE
+#define _MULTIARRAYMODULE
+
+#define PY_SSIZE_T_CLEAN
+#include <Python.h>
+#include <structmember.h>
+
+#include "numpy/ndarraytypes.h"
+#include "numpy/npy_common.h"
+#include "numpy/arrayobject.h"
+#include "npy_import.h"
+#include "npy_static_data.h"
+
+// static variables are zero-filled by default, no need to explicitly do so
+NPY_VISIBILITY_HIDDEN npy_interned_str_struct npy_interned_str;
+NPY_VISIBILITY_HIDDEN npy_static_pydata_struct npy_static_pydata;
+NPY_VISIBILITY_HIDDEN npy_static_cdata_struct npy_static_cdata;
+
+NPY_NO_EXPORT int
+intern_strings(void)
+{
+    npy_interned_str.current_allocator = PyUnicode_InternFromString("current_allocator");
```

Comment:
I think there are ways to make this more compact, but maybe not worth the effort. A different paradigm (for instance, making `npy_interned_str` an array of pointers, and using enum-ed constants like `CURRENT_ALLOCATOR` to index the array) might allow checking that all the values/fields have been initialized. Here I guess you could do a loop over the `sizeof(npy_interned_str)/sizeof(char *)` at the end of the function and check for `NULL`

Likewise, it would be nice to check `npy_static_pydata` for NULL values after initialization.

Reply:
See the latest commit which adds a function that verifies all entries in the two structs are non-null once module initialization is complete.

It unfortunately can't happen right after initializing the structs in this file here because some items (e.g. the reference to ndarray's `__array_ufunc__` implementation) cannot be created until relatively late during module initialization, but other things are needed early.

Reply:
Thanks, I think that check will prevent future problems when changing these structs.